features/net/network-socket/NetworkInterface.h

```diff
@@ -94,6 +94,22 @@ class NetworkInterface {
      */
     virtual int connect() = 0;
 
+    /** Sends DHCP request
+     *
+     * @param timeout Request timeout in ms
+     * @return NSAPI_ERROR_OK in case of success, error code otherwise
+     */
+    virtual int start_dhcp(unsigned int timeout = 15000) = 0;
```

Comment:
The latest rebase brought this dhcp changes in, looks like that's what we needed. I'll remove this.